flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/TableEnvironment.scala

```diff
@@ -498,6 +505,10 @@ abstract class TableEnvironment(val config: TableConfig) {
     TableEnvironment.validateType(inputType)
```

Comment:
The `validateType()` method is also used to check the output type, i.e., the type of a DataSet or DataStream which is created from a Table. I think we should allow `GenericType<Row>` in that case.
Hence, I would keep the `validateType()` method as it is.